src/php/ext/grpc/channel.c

```diff
@@ -148,37 +148,33 @@ PHP_METHOD(Channel, __construct) {
                          "Channel expects a string and an array", 1 TSRMLS_CC);
     return;
   }
-  if (args_array == NULL) {
```

Comment:
According to the code a few lines up, `args_array` is an optional argument, so it can be `NULL`. Please either change that or revert this.